ArchiSteamFarm/ArchiWebHandler.cs

```diff
@@ -122,39 +122,52 @@ public sealed class ArchiWebHandler : IDisposable {
 		}
 
 		[ItemCanBeNull]
+		[Obsolete]
 		[PublicAPI]
-		[SuppressMessage("ReSharper", "FunctionComplexityOverflow")]
-		public async Task<HashSet<Steam.Asset>> GetInventory(ulong steamID = 0, uint appID = Steam.Asset.SteamAppID, ulong contextID = Steam.Asset.SteamCommunityContextID, bool? marketable = null, bool? tradable = null, IReadOnlyCollection<uint> wantedRealAppIDs = null, IReadOnlyCollection<uint> unwantedRealAppIDs = null, IReadOnlyCollection<Steam.Asset.EType> wantedTypes = null, IReadOnlyCollection<(uint RealAppID, Steam.Asset.EType Type, Steam.Asset.ERarity Rarity)> wantedSets = null) {
+		public async Task<HashSet<Steam.Asset>> GetInventory(ulong steamID = 0, uint appID = Steam.Asset.SteamAppID, ulong contextID = Steam.Asset.SteamCommunityContextID) {
 			if ((appID == 0) || (contextID == 0)) {
 				Bot.ArchiLogger.LogNullError(nameof(appID) + " || " + nameof(contextID));
 
 				return null;
 			}
 
+			try {
+				return await GetInventoryEnumerable(steamID, appID, contextID).ToHashSetAsync().ConfigureAwait(false);
+			} catch (Exception e) {
+				Bot.ArchiLogger.LogGenericWarning(string.Format(Strings.WarningFailedWithError, e.Message));
+
+				return null;
+			}
+		}
+
+		[PublicAPI]
+		[SuppressMessage("ReSharper", "FunctionComplexityOverflow")]
+		public async IAsyncEnumerable<Steam.Asset> GetInventoryEnumerable(ulong steamID = 0, uint appID = Steam.Asset.SteamAppID, ulong contextID = Steam.Asset.SteamCommunityContextID) {
+			if ((appID == 0) || (contextID == 0)) {
+				throw new Exception(string.Format(Strings.ErrorObjectIsNull, nameof(appID) + " || " + nameof(contextID)));
+			}
+
 			if (steamID == 0) {
 				if (!Initialized) {
 					for (byte i = 0; (i < ASF.GlobalConfig.ConnectionTimeout) && !Initialized && Bot.IsConnectedAndLoggedOn; i++) {
 						await Task.Delay(1000).ConfigureAwait(false);
 					}
 
 					if (!Initialized) {
-						Bot.ArchiLogger.LogGenericWarning(Strings.WarningFailed);
-
-						return null;
+						throw new Exception(Strings.WarningFailed);
```

Comment:
Don't we have a better exception type? I think throwing (and catching) `IOException` is more proper for us, we don't want to just catch all exceptions and handle them as network failures, there might be some code problem which we want to be notified about.